### ISSUE TITLE
Support for ND-Dilated Convolutions

### ISSUE THREAD

**Issue by MannyKayy** [chainer/chainer#2843]
Chainer seems to only support 2D dilated convolutions. It would be nice to have ND dilated convolutions. This would be handy for work in other domains such as text, audio and data of other modalities. Some frameworks have now included a dilation argument within the convolution layer (along with the traditional things such as padding, stride, etc..).

Refs:
[Caffe Support](https://github.com/BVLC/caffe/pull/3452)
[Keras Support](https://keras.io/layers/convolutional)

**Comment by MannyKayy** [comment]
Bumping to keep this issue open ..

**Comment by beam2d** [comment]
The issue still exists.

**Comment by kmaehashi** [comment]
PR ongoing: #4591
